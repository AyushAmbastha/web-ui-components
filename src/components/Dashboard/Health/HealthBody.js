import React from 'react';
import PropTypes from 'prop-types';

export const HealthBody = ({ children }) => <div className="kubevirt-dashboard-health__items">{children}</div>;

HealthBody.propTypes = {
  children: PropTypes.oneOfType([PropTypes.node, PropTypes.func]).isRequired,
};
