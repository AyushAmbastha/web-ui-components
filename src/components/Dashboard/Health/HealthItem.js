import React from 'react';
import PropTypes from 'prop-types';
import { Icon } from 'patternfly-react';

export const HealthItem = ({ icon, classname, message }) => (
  <div className="kubevirt-health__body">
    <Icon type="fa" size="2x" name={icon} className={`kubevirt-health__icon kubevirt-health__icon--${classname}`} />
    <span className="kubevirt-health__text">{message}</span>
  </div>
);

HealthItem.propTypes = {
  message: PropTypes.string.isRequired,
  icon: PropTypes.string.isRequired,
  classname: PropTypes.string.isRequired,
};
