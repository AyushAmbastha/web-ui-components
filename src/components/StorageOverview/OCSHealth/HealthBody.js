import React from 'react';
import PropTypes from 'prop-types';
import { Icon } from 'patternfly-react';

const healtyIcon = (
  <div className="ocs-health__icon--ok">
    <Icon type="fa" size="2x" name="check-circle" />
  </div>
);

const errorIcon = (
  <div className="ocs-health__icon--error">
    <Icon type="fa" size="2x" name="exclamation-circle" />
  </div>
);

const warningIcon = (
  <div className="ocs-health__icon--warning">
    <Icon type="fa" size="2x" name="exclamation-triangle" />
  </div>
);

const OCSHealthMap = {
  '0': {
    icon: healtyIcon,
    message: 'OCS is Healthy',
  },
  '1': {
    icon: warningIcon,
    message: 'OCS is in Warning',
  },
  '2': {
    icon: errorIcon,
    message: 'OCS is in Error',
  },
};

const HealthBody = ({ data }) => (
  <React.Fragment>
    {OCSHealthMap[data.healthy].icon}
    <span className="ocs-ocshealth__row-status-item-text">{OCSHealthMap[data.healthy].message}</span>
  </React.Fragment>
);

HealthBody.propTypes = {
  data: PropTypes.object.isRequired,
};

export default HealthBody;
