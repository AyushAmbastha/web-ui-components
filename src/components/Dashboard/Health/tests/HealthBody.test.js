import React from 'react';
import { shallow } from 'enzyme';

import { HealthBody } from '../HealthBody';
import { default as HealthItemFixture } from '../fixtures/HealthItem.fixture';
import { HealthItem } from '../HealthItem';

const testHealthBody = () => (
  <HealthBody>
    <HealthItem {...HealthItemFixture[0].props} />
  </HealthBody>
);

describe('<HealthBody />', () => {
  it('renders correctly', () => {
    const component = shallow(testHealthBody());
    expect(component).toMatchSnapshot();
  });
});
