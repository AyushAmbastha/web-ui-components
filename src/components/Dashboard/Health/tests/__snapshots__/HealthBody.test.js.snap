// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HealthBody /> renders correctly 1`] = `
<div
  className="kubevirt-health__items"
>
  <HealthItem
    classname="error"
    data={
      Object {
        "healthy": Object {
          "classname": "ok",
          "iconname": "check-circle",
          "message": "OCS is Healthy",
        },
      }
    }
    icon="exclamation-triangle"
    message="Cant get health"
  />
</div>
`;
